implementations/elixir/ockam/ockam/lib/ockam/protocol/bare_extended.ex

```diff
@@ -71,3 +73,79 @@ defmodule Ockam.Bare.Extended do
     extended_schema
   end
 end
+
+
+defmodule Ockam.Bare.Variant do
+  @moduledoc """
+  Support variant types
+  Variant types are defined as {:variant, [atom() | {atom(), schema()]}
```

Comment:
Is that how Rust implementation encodes it?
I don't see it in the BARE spec. It looks like a union with `:void` type.
This notation looks more clean, but is not documented in BARE.
We'll have to contribute it to upstream at some point, but for now it's ok.

Reply:
Right, I didn't found it documented neither.  Learn it by looking at the data that was received from the rust side.

See here:
```
    let ping = PortalMessage::Ping;
    let pong = PortalMessage::Pong;
    let disc = PortalMessage::Disconnect;
    let payload = PortalMessage::Payload(vec![55, 55, 55, 55]);

    println!("Ping: {:?}", serde_bare::to_vec(&ping));
    println!("Pong: {:?}", serde_bare::to_vec(&pong));
    println!("Disc: {:?}", serde_bare::to_vec(&disc));
    println!("Payload: {:?}", serde_bare::to_vec(&payload));

Ping: Ok([0])
Pong: Ok([1])
Disc: Ok([2])
Payload: Ok([3, 4, 55, 55, 55, 55])
```
The implementation is more like a workaround that serve for the current message definitions we have, but it won't handle cases where the variant is not the top-most definition.   Not sure if upstreaming this is an option given that seems to be an addition/deviation of the rust implementation.   Perhaps we can get by with this till https://github.com/ockam-network/ockam/issues/2496 is defined.